include/freerdp/api.h

```diff
@@ -66,5 +66,6 @@
 
 #define IFCALL(_cb, ...) do { if (_cb != NULL) { _cb( __VA_ARGS__ ); } } while (0)
 #define IFCALLRET(_cb, _ret, ...) do { if (_cb != NULL) { _ret = _cb( __VA_ARGS__ ); } } while (0)
+#define IFCALLRETURN(_default_return, _cb, ...) do { if (_cb != NULL) { return _cb( __VA_ARGS__ ); } else { return _default_return;} } while (0)
```

Comment:
IMHO It is not really obvious in the code (unless studying the macro of course) that IFCALLRETURN actually returns and maybe the name has to little contrast to IFCALLRET.
I'd suggest to change it to something like this:
```
#define IFCALLASSIGN(_default_return, _cb, ...)  (_cb != NULL) ? _cb( __VA_ARGS__ ) : (_default_return)
```
Then it would be clearer in the code because of the explicit return:
e.g. ``` return IFCALLASSIGN(-1, addThreeValues, 4, 5, 6)```

just my two ¢ ;)